Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Motion/MotionDataBuilder.cpp

```diff
@@ -220,8 +248,10 @@ namespace EMotionFX
                     continue;
                 }
 
+                // Check if we are dealing with a transform node or a bone and only recurse down the node hierarchy in this case.
                 AZStd::shared_ptr<const SceneDataTypes::IBoneData> nodeBone = azrtti_cast<const SceneDataTypes::IBoneData*>(it->second);
-                if (!nodeBone)
+                AZStd::shared_ptr<const SceneDataTypes::ITransform> nodeTransform = azrtti_cast<const SceneDataTypes::ITransform*>(it->second);
```

Comment:
Only one of these will ever be set to a valid value, but either way one of them will unnecessarily increment the use count. We decay them to raw pointers where they are used on line 322 anyway.

Reply:
Changed